opcodes/dxil/dxil_nvapi.cpp

```diff
@@ -487,6 +487,57 @@ static bool emit_nvapi_extn_op_hit_object_get_ray_desc(Converter::Impl &impl)
 	return true;
 }
 
+static bool emit_nvapi_extn_op_hit_object_load_local_root_table_constant(Converter::Impl &impl)
+{
+	spv::Id hit_object = get_argument(impl, NVAPI_ARGUMENT_SRC0U + 0);
+	spv::Id offset = get_argument(impl, NVAPI_ARGUMENT_SRC0U + 1);
+
+	auto &builder = impl.builder();
+
+	spv::Id uint32 = builder.makeUintType(32);
+	spv::Id uint64 = builder.makeUintType(64);
+
+	auto *op = impl.allocate(spv::OpHitObjectGetShaderRecordBufferHandleNV, builder.makeVectorType(uint32, 2));
+	op->add_id(hit_object);
+	impl.add(op);
+
+	auto *convert_op = impl.allocate(spv::OpUConvert, builder.makeVectorType(uint64, 2));
+	convert_op->add_id(op->id);
+	impl.add(convert_op);
+
+	auto *extract_op = impl.allocate(spv::OpCompositeExtract, uint64);
+	extract_op->add_id(convert_op->id);
+	extract_op->add_literal(0);
+	impl.add(extract_op);
+
+	convert_op = impl.allocate(spv::OpUConvert, uint64);
+	convert_op->add_id(offset);
+	impl.add(convert_op);
+
+	auto *add_op = impl.allocate(spv::OpIAdd, uint64);
+	add_op->add_id(extract_op->id);
+	add_op->add_id(convert_op->id);
+	impl.add(add_op);
+
+	convert_op = impl.allocate(spv::OpConvertUToPtr, impl.nvapi.hit_object_srb_pointer_type_id);
```

Comment:
You can just synthesize the block type here and cache it in impl.nvapi. No need to pollute core code.

Reply:
Huh, I thought it would be too late for that but indeed it doesn't seem to be.